File: DomainObjects/Account/Goal.cs
using Auctus.Util.DapperAttributes;
using System;
using System.Collections.Generic;
using System.Text;

namespace Auctus.DomainObjects.Account
{
    public class Goal
    {
        [DapperKey(true)]
        [DapperType(System.Data.DbType.Int32)]
        public int Id { get; set; }
        [DapperType(System.Data.DbType.Int32)]
        public int UserId { get; set; }
        [DapperType(System.Data.DbType.DateTime)]
        public DateTime CreationDate { get; set; }
        [DapperType(System.Data.DbType.Decimal)]
        public double? TargetAmount { get; set; }
        [DapperType(System.Data.DbType.Decimal)]
        public double? StartingAmount { get; set; }
        [DapperType(System.Data.DbType.Decimal)]
        public double? MonthlyContribution { get; set; }
        [DapperType(System.Data.DbType.Int32)]
        public int? Timeframe { get; set; }
        [DapperType(System.Data.DbType.Int32)]
        public int? Risk { get; set; }
        [DapperType(System.Data.DbType.Int32)]
        public int GoalOptionId { get; set; }

        public GoalOption GoalOption { get; set; }
    }
}

File: Web/src/app/model/account/goal.ts
import { GoalOption } from "./goalOption";

export class Goal {

  id: number;
  targetAmount: number;
  startingAmount: number;
  monthlyContribution: number;
  timeframe: number;
  risk: number;
  goalOption: GoalOption;
  
  constructor() {
    this.startingAmount = 10000;
    this.monthlyContribution = 300;
    this.timeframe = 30;
    this.risk = 3;
    this.goalOption = new GoalOption();
  }
}
